dotnet/samples/Concepts/TextGeneration/HuggingFace_TextGeneration.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Connectors.HuggingFace;
+using xRetry;
+
+#pragma warning disable format // Format item can be simplified
+#pragma warning disable CA1861 // Avoid constant arrays as arguments
+
+namespace Examples;
+
+// The following example shows how to use Semantic Kernel with HuggingFace API.
+public class HuggingFace_TextGeneration(ITestOutputHelper helper) : BaseTest(helper)
+{
+    /// <summary>
+    /// This example uses HuggingFace Inference API to access hosted models.
+    /// More information here: <see href="https://huggingface.co/inference-api"/>
+    /// </summary>
+    [Fact]
+    public async Task RunInferenceApiExampleAsync()
+    {
+        WriteLine("\n======== HuggingFace Inference API example ========\n");
+
+        Kernel kernel = Kernel.CreateBuilder()
+            .AddHuggingFaceTextGeneration(
+                model: TestConfiguration.HuggingFace.ModelId,
+                apiKey: TestConfiguration.HuggingFace.ApiKey)
+            .Build();
+
+        var questionAnswerFunction = kernel.CreateFunctionFromPrompt("Question: {{$input}}; Answer:");
+
+        var result = await kernel.InvokeAsync(questionAnswerFunction, new() { ["input"] = "What is New York?" });
```

Comment:
In my recent tests I always had to set the `max_new_tokens` to get a valid response. Why is this not needed here?

Reply:
It depends on the model configured.

Using: `HuggingFaceH4/zephyr-7b-beta` I get this output.
![image](https://github.com/microsoft/semantic-kernel/assets/19890735/1ab936c2-0f0c-485d-9e92-7470381f4919)

Added the remarks with details about the tested as well as defaulting to one of the successful tested models.